services/blockstorage/service.go

```diff
@@ -62,12 +63,91 @@ func (s *service) CommitBlock(input *services.CommitBlockInput) (*services.Commi
 	return nil, nil
 }
 
+func (s *service) lookForTransactionsBlockHeader(height primitives.BlockHeight) (*services.GetTransactionsBlockHeaderOutput, error) {
+	txBlock, err := s.persistence.GetTransactionsBlock(height)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &services.GetTransactionsBlockHeaderOutput{
+		TransactionsBlockProof: txBlock.BlockProof,
+		TransactionsBlockHeader: txBlock.Header,
+		TransactionsBlockMetadata: txBlock.Metadata,
+	}, nil
+}
+
 func (s *service) GetTransactionsBlockHeader(input *services.GetTransactionsBlockHeaderInput) (*services.GetTransactionsBlockHeaderOutput, error) {
-	panic("Not implemented")
+	if input.BlockHeight > s.lastCommittedBlockHeight && input.BlockHeight - s.lastCommittedBlockHeight <= 5 {
+		c := make(chan *services.GetTransactionsBlockHeaderOutput)
+
+		go func() {
+			// TODO extract to a config
+			const interval = 10
+			const timeout = 10000
+
+			for i:=0; i < timeout; i+= interval {
+				if input.BlockHeight <= s.lastCommittedBlockHeight {
+					lookupResult, err := s.lookForTransactionsBlockHeader(input.BlockHeight)
+
+					if err == nil {
```

Comment:
is this good enough? I think there should be a distinction between error types. Should all errors be retried?

Reply:
We retry until it times out, which is expected behavior from the spec.

Reply:
I'm not sure this is well-defined in the spec. Should you retry on unrecoverable errors?